app/helpers/close_contact_query_helper.rb

```diff
@@ -5,4 +5,110 @@ module CloseContactQueryHelper
+  # Queries close_contact by ID or type based on search text.
+  def search(close_contact, search)
+    return close_contact if search.blank?
+
+    close_contact.where('id like ?', "#{search&.downcase}%").or(
+      close_contact.where('first_name like ?', "#{search&.downcase}%").or(
+        close_contact.where('last_name like ?', "#{search&.downcase}%").or(
+          close_contact.where('primary_telephone like ?', Phonelib.parse(search, 'US').full_e164).or(
+            close_contact.where('email like ?', "#{search&.downcase}%").or(
+              close_contact.where('last_date_of_exposure like ?', "#{search&.downcase}%").or(
+                close_contact.where('assigned_user like ?', "#{search&.downcase}%").or(
+                  close_contact.where('contact_attempts like ?', "#{search&.downcase}%").or(
+                    close_contact.where('notes like ?', "#{search&.downcase}%")
+                  )
+                )
+              )
+            )
+          )
+        )
+      )
+    )
+  end
+
+  # Sorts close_contacts based on given column and direction.
+  def sort(close_contacts, order, direction)
+    # Order by created_at date by default
+    return close_contacts.order(created_at: 'desc') if order.blank? || direction.blank?
+
+    # Satisfy brakeman with additional sanitation logic
+    dir = direction == 'asc' ? 'asc' : 'desc'
+
+    case order
+    when 'first_name'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN first_name IS NULL THEN 1 ELSE 0 END, first_name ' + dir))
+    when 'last_name'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN last_name IS NULL THEN 1 ELSE 0 END, last_name ' + dir))
+    when 'primary_telephone'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN primary_telephone IS NULL THEN 1 ELSE 0 END, primary_telephone ' + dir))
+    when 'email'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN email IS NULL THEN 1 ELSE 0 END, email ' + dir))
+    when 'enrolled_id'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN enrolled_id IS NULL THEN 1 ELSE 0 END, enrolled_id ' + dir))
+    when 'last_date_of_exposure'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN last_date_of_exposure IS NULL THEN 1 ELSE 0 END, last_date_of_exposure ' + dir))
+    when 'assigned_user'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN assigned_user IS NULL THEN 1 ELSE 0 END, assigned_user ' + dir))
+    when 'contact_attempts'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN contact_attempts IS NULL THEN 1 ELSE 0 END, contact_attempts ' + dir))
+    when 'notes'
+      close_contacts = close_contacts.order(Arel.sql("CASE WHEN notes IS NULL THEN 2 WHEN notes = '' THEN 1 ELSE 0 END, notes " + dir))
+    end
+    close_contacts
+  end
+
+  # Paginates close_contact data.
+  def paginate(close_contacts, entries, page)
+    return close_contacts if entries.blank? || entries <= 0 || page.blank? || page.negative?
```

Comment:
can you also make this change in the `vaccine_query_helper`

```suggestion
    return close_contacts if entries.blank? || entries.negative? || page.blank? || page.negative?
```

Reply:
I copy and pasted this from elsewhere, but sounds good! Will change it in all locations

Reply:
correction: `<= 0` is not the same as negative. Reverting the change.